Updates to the examples.

Alex-Ian-Hamilton opened the pull request.

Alex-Ian-Hamilton committed:
Added examples for extrapolating and visualising potential fields from HMI (using VSO) and example generated data.
Added more documentation and various tweaks/fixes to a number of functions.
Added the ability for extrapolators to crop/submap as defined by the user.
Added units support (notably percentage positions and astropy quantity ranges) to the example data generator.

Alex-Ian-Hamilton committed:
Working on making the examples work.

Alex-Ian-Hamilton committed:
Merge branch 'master' of https://github.com/sunpy/solarbextrapolation

Alex-Ian-Hamilton committed:
More on example.

Alex-Ian-Hamilton committed:
Examples in the correct location and working.

Alex-Ian-Hamilton committed:
Small tweaks to the examples.

Alex-Ian-Hamilton committed:
Removed __name__ '__main__' from example files.

Alex-Ian-Hamilton committed:
Tweaked classes to force matplotlib (mpl) not to use Qt for it's backend.
This fixes a bug where MayaVi and mpl both try to use Qt and we get the error:
"ValueError: API 'QString' has already been set to version 1"

Alex-Ian-Hamilton committed:
Added numba support to the potential extrapolator (with enable/disable option).
Added the ability for the extrapolate method to pass down kwargs, allowing a level of user control.

Cadair commented on solarbextrapolation/classes.py:
```diff
@@ -535,10 +553,9 @@ def compare_all(self):
 #
 #######
 if __name__ == '__main__':
```
This should not be here anymore.

Cadair commented on solarbextrapolation/example_full_potential_extrapolation.py:
```diff
@@ -3,6 +3,9 @@
 Created on Mon Aug 10 11:26:13 2015
 
 @author: alex_
+
```
should this file now be in `examples/`?

Cadair commented on solarbextrapolation/example_generate_example_data.py:
```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+"""
```
same with this one? Should it be in `examples/`?

Cadair commented on solarbextrapolation/potential_field_extrapolator.py:
```diff
@@ -149,13 +84,36 @@ def _extrapolate_phi(self, debug=False):
+            try:
+                import numba
+                from potential_field_extrapolator_numba import *
+                return phi_extrapolation_numba(arr_boundary, self.shape, self.Dx.value, self.Dy.value, self.Dz.value)
```
rather than repeating the call twice I would do this:

``` python
if enable_numba:
    try:
        import numba
        from potential_field_extrapolator_numba import phi_extrapolation_numba as phi_extrapolation
    except ImportError:
        from potential_field_extrapolator_python import phi_extrapolation_python as phi_extrapolation
return phi_extrapolation(arr_boundary, self.shape, self.Dx.value, self.Dy.value, self.Dz.value)
```

Cadair commented on solarbextrapolation/potential_field_extrapolator_numba.py:
```diff
@@ -0,0 +1,81 @@
+# Using numba/JIT for speeding up.
+Gn_5_2_29_numba = autojit(Gn_5_2_29)
+
+def phi_extrapolation_python(boundary, shape, Dx, Dy, Dz):
```
rather than copying these from here, just import them from the python version?

Cadair commented on solarbextrapolation/potential_field_extrapolator_numba.py:
```diff
@@ -0,0 +1,81 @@
+    return floOut
+
+# Using numba/JIT for speeding up.
+Gn_5_2_29_numba = autojit(Gn_5_2_29)
```
does having this jitted make a lot of difference?

Cadair commented on solarbextrapolation/potential_field_extrapolator_numba.py:
```diff
@@ -0,0 +1,81 @@
+                        
+                        # Find the components for this contribution product
+                        B_n = boundary[i_prime, j_prime]
+                        G_n = Gn_5_2_29_numba(x, y, z, xP, yP, DxDy, z_submerge)
```
to save you having to duplicate this whole function to change this one line, you could make the Python version take this function as an argument, then you could pass in the jitted version.

Alex-Ian-Hamilton committed:
Tweaks on the potential extrapolator.
Added performance tyests example to compare extrapolator runtimes.
Added example for showing image to potential extrapolator, but this doesn't work so well.

Alex-Ian-Hamilton committed:
Various updates, added some work to the anylitical model class and the comparer.
Updated examples and dummy data generator.
